src/components/billingRecord/IFXBillingRecordList.vue

```diff
@@ -292,20 +295,24 @@ export default {
           missing.push(i)
         }
       }
-      Promise.all(missing.map((i) => me.getFullBillingRecordByItemIndex(i)))
-        .then(() => {
-          const toBeUpdated = []
-          me.items.forEach((it1) => {
-            items.forEach((it2) => {
-              if (it1.id === it2.id) {
-                it1.billingRecordStates.push({ name: state, user: '', approvers: [], comment: '' })
-                toBeUpdated.push(it1)
-              }
-            })
+      const promises = missing.map((i) => me.getFullBillingRecordByItemIndex(i))
+      while (promises.length) {
+        await new Promise(r => setTimeout(r, 500));
```

Comment:
Why are you batching these promises up and waiting between each? Does the backend get overwhelmed or something? Or is there supposed to be a progress bar of some kind?

Reply:
Mostly there would be 20-30.  A big batch of RC could be many hundreds.  Could be a bit much.

Reply:
You're firing them all off at once but then waiting in batches; that's the part I don't really understand. Do you want to show a progress indicator or something between waits? Or is the `sleep` there to allow user interaction?